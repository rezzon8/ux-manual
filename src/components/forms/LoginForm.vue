<template>
  <v-form @submit.prevent>
    <v-text-field
      v-model="email"
      :error-messages="emailErrors"
      label="Email"
      required
      @input="$v.email.$touch()"
      @blur="$v.email.$touch()"
    ></v-text-field>
    <v-text-field
      v-model="password"
      :error-messages="passwordErrors"
      counter="16"
      label="Password"
      required
      :append-icon="showPassword ? 'mdi-eye' : 'mdi-eye-off'"
      @click:append="showPassword = !showPassword"
      :type="showPassword ? 'text' : 'password'"
      @input="$v.password.$touch()"
      @blur="$v.password.$touch()"
    ></v-text-field>

    <v-btn
      :disabled="isFormSubmitReady"
      class="mr-4 "
      type="submit"
      @click.native="logIn"
      :loading="$store.state.is_data_processing"
    >
      Login
    </v-btn>
    <v-btn text @click.native="clearForm" class="clear-form">
      clear
    </v-btn>
  </v-form>
</template>

<script>
import { validationMixin } from "vuelidate";
import { required, email } from "vuelidate/lib/validators";

export default {
  name: "LoginForm",

  mixins: [validationMixin],

  validations: {
    email: { required, email },
    password: { required }
  },

  data: () => ({
    email: "",
    password: "",
    showPassword: false
  }),

  computed: {
    emailErrors() {
      const errors = [];
      if (!this.$v.email.$dirty) return errors;
      !this.$v.email.email && errors.push("Must be valid e-mail.");
      !this.$v.email.required && errors.push("E-mail is required.");
      return errors;
    },
    passwordErrors() {
      const errors = [];
      if (!this.$v.password.$dirty) return errors;
      !this.$v.password.required && errors.push("Password is required.");
      return errors;
    },
    isFormSubmitReady() {
      return this.$v.$invalid;
    }
  },

  methods: {
    logIn() {
      this.$v.$touch();
      if (!this.$v.$invalid) {
        const payload = {
          email: this.email,
          password: this.password
        };
        this.$emit("login", payload);
      }
    },
    clearForm() {
      this.$v.$reset();
      this.email = "";
      this.password = "";
      this.showPassword = false;
    }
  }
};
</script>

<style></style>
