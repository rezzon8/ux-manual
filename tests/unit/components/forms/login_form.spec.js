import { createLocalVue, mount } from "@vue/test-utils";
import Vue from "vue";
import Vuetify from "vuetify";
import LoginForm from "@/components/forms/LoginForm.vue";

Vue.use(Vuetify); // added to prevent vuetify lint errors

const localVue = createLocalVue();
const vuetify = new Vuetify();

let wrapper;
// added to prevent Vuetify error
document.body.setAttribute("data-app", true);

describe("LoginForm.vue", () => {
  beforeEach(() => {
    wrapper = mount(LoginForm, {
      localVue,
      vuetify
    });
  });
  it("should render", () => {
    console.log(wrapper.html());
    expect(wrapper.exists()).toBe(true);
  });
});
