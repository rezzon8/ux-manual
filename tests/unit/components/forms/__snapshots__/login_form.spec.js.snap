// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.vue should render 1`] = `
<form novalidate="novalidate" class="v-form">
  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label for="input-13" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Email</label><input required="required" id="input-13" type="text"></div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
        </div>
      </div>
    </div>
  </div>
  <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-text-field__slot"><label for="input-16" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Password</label><input required="required" id="input-16" type="password"></div>
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><button type="button" aria-label="append icon" class="v-icon notranslate v-icon--link mdi mdi-eye-off theme--light"></button></div>
        </div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light">
          <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
        </div>
        <div class="v-counter theme--light">8 / 16</div>
      </div>
    </div>
  </div> <button type="submit" class="mr-4  v-btn v-btn--contained v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">
    Login
  </span></button> <button type="button" class="clear-form v-btn v-btn--flat v-btn--text theme--light v-size--default"><span class="v-btn__content">
    clear
  </span></button>
</form>
`;
